Adds a python script that will create a CSV file with dataset provenance information

cmadjar opened the pull request:
## Description

This is the second time now that I have to provide a spreadsheet summary of the datasets provenance for reporting purposes. Now that we have the wonderful `origin` fields in the `extraProperties` of the DATS.json files for almost all datasets, a script could be created to generate such a spreadsheet (CSV).

To create the CSV file with the dataset provenance summary, execute the `create_data_provenance_summary.py` script with option `-d <path to conp-dataset>`. Running the script with option `-h` will display how to use the script

```bash
$ python create_data_provenance_summary.py -h

This tool facilitates the aggregation of data provenance for reporting purposes. It will read DATS files and print out a summary of data provenance based on the `origin` fields present in the DATS.json files of every dataset present in conp-dataset directory.

usage  : python create_data_provenance_summary.py -d <conp-dataset directory path>

options: 
	-d: path to the conp-dataset directory to parse
```

Once the script has finished running, it will have created a `dataset_provenance_<date>.csv` which will contain the following information for example:
| Dataset                                                                                                                                             | Principal Investigator | Consortium                                 | Institution                                   | City      | Province         | Country        |
|-----------------------------------------------------------------------------------------------------------------------------------------------------|------------------------|--------------------------------------------|-----------------------------------------------|-----------|------------------|----------------|
| BigBrain dataset                                                                                                                                    |                        | BigBrain project                           |                                               |           |                  |                |
| PREVENT-AD open data in BIDS format                                                                                                                 |                        | PREVENT-AD Research Group                  |                                               | Montreal  | Quebec           | Canada         |
| Multimodal data with wide-field GCaMP imaging                                                                                                       |                        |                                            |                                               |           |                  |                |
| BigBrain dataset - 3D Surfaces (derived dataset)                                                                                                    |                        | BigBrain project                           |                                               |           |                  |                |
| Intracellular Recordings of Murine Neocortical Neurons                                                                                              |                        |                                            |                                               |           |                  |                |
| McGill EMC: RNA-seq experiment from human brain sample                                                                                              |                        |                                            | McGill EMC                                    | Montreal  | Quebec           | Canada         |
| Neurodevelopmental Disorders Priority Setting Partnership                                                                                           |                        |                                            | Ontario Brain Institute                       | Toronto   | Ontario          | Canada         |
| OpenPain cbp_resting                                                                                                                                |                        |                                            | Northwestern University                       | Evanston  | Illinois         | USA            |
| OpenPain BrainNetworkChange_Mano                                                                                                                    |                        |                                            | USA                                           | Evanston  | Illinois         |                |
| OpenPain subacute_longitudinal_study                                                                                                                |                        |                                            | Northwestern University                       | Evanston  | Ontario          | USA            |
| OpenPain thermal                                                                                                                                    |                        |                                            | Northwestern University                       | Evanston  | Illinois         | USA            |
| OpenPain placebo_1                                                                                                                                  |                        |                                            | Northwestern University                       | Evanston  | Illinois         | USA            |
| OpenPain placebo_predict_tetreault                                                                                                                  |                        |                                            | Northwestern University                       | Evanston  | Illinois         | USA            |
| Systematic phenomics analysis of autism-associated genes reveals parallel networks underlying reversible impairments in habituation                 |                        |                                            | University of British Columbia                | Vancouver | British Columbia | Canada         |
| High Resolution Magnetic Resonance Imaging of Mouse Model related to Autism                                                                         |                        |                                            |                                               |           |                  |                |
| Multicenter Single Subject Human MRI Phantom                                                                                                        | Alan C. Evans          |                                            |                                               |           |                  |                |
| Visual working memory: Study of Task fMRI and Behavioural response                                                                                  |                        |                                            | Imperial College London                       | London    |                  | United Kingdom |
| A Transcriptomic Atlas of Mouse Neocortical Layers                                                                                                  | Chris P. Ponting       |                                            | Oxford University                             | Oxford    |                  | United Kingdom |
| Homo sapiens reference genome sequences                                                                                                             |                        | Genome Reference Consortium                |                                               |           |                  |                |
| Mus musculus reference genome sequence                                                                                                              |                        | Genome Reference Consortium                |                                               |           |                  |                |
| Callithrix jacchus reference sequence                                                                                                               |                        |                                            | National Center for Biotechnology Information | Bethesda  | Maryland         | USA            |
| MRI and unbiased averages of wild muskrats (Ondatra zibethicus) and red squirrels (Tamiasciurus hudsonicus)                                         |                        |                                            |                                               |           |                  |                |
| BrainSpan: Atlas of the Developing Human Brain                                                                                                      |                        |                                            | The Allen Institute for Brain Science         | Seattle   | Washington       | USA            |
| PREVENT-AD open data                                                                                                                                |                        | PREVENT-AD Research Group                  |                                               | Montreal  | Quebec           | Canada         |
| Sharing and reusing gene expression profiling data in neuroscience                                                                                  |                        |                                            | University of British Columbia (UBC)          | Vancouver | British Columbia | Canada         |
| cneuromod dataset                                                                                                                                   |                        | The Courtois Project on Neuronal Modelling |                                               | Montreal  | Quebec           | Canada         |
| 1000 Genomes Project                                                                                                                                |                        | 1000 Genomes Project                       |                                               |           |                  |                |
| Comparing Perturbation Modes for Evaluating Instabilities in Neuroimaging: Processed NKI-RS Subset (08/2019)                                        |                        |                                            |                                               |           |                  |                |
| Long-term, Phantom Quality Assurance data for functional Magnetic Resonance Imaging from 13 3T research scanners                                    |                        |                                            | Ontario Brain Institute                       | Toronto   | Ontario          | Canada         |
| Hippocampal morphology and cytoarchitecture in the 3D BigBrain                                                                                      |                        |                                            | University of Western Ontario                 | London    | Ontario          | Canada         |
| HCPUR100: Healthy Adult human Brain Diffusion Template                                                                                              |                        |                                            | University of Western Ontario                 | London    | Ontario          | Canada         |
| Accurate registration of the BigBrain dataset with the MNI PD25 and ICBM152 atlases                                                                 |                        |                                            | University of Western Ontario                 | London    | Ontario          | Canada         |
| Multi-model functionalization of disease-associated PTEN missense mutations identifies multiple molecular mechanisms underlying protein dysfunction |                        |                                            | University of British Columbia                | Vancouver | British Columbia | Canada         |
| The Cuban Human Brain Mapping Normative database (EEG, MRI, and Cognition dataset)                                                                  | Pedro A. Valdes-Sosa   |                                            | Cuban Neuroscience Center                     | Havana    |                  | Cuba           |
| Allen Brain Atlas: Cell Types                                                                                                                       |                        |                                            | The Allen Institute for Brain Science         | Seattle   | Washington       | USA            |
| PERFORM Dataset; one control subject                                                                                                                |                        |                                            |                                               |           |                  |                |
| NIH Blueprint Non-Human Primate Atlas                                                                                                               |                        |                                            | The Allen Institute for Brain Science         | Seattle   | Washington       | USA            |
| SIMON                                                                                                                                               |                        |                                            |                                               |           |                  |                |
| Learning Naturalistic Structure: Processed fMRI dataset                                                                                             |                        |                                            |                                               |           |                  |                |
| VFA T1 mapping \| RTHawk (open) vs Siemens (commercial)                                                                                             |                        |                                            |                                               |           |                  |                |
| BigBrain dataset - 3D Classified Volumes (derived dataset)                                                                                          |                        | BigBrain project                           |                                               |           |                  |                |

cmadjar commented:
@glatard @emmetaobrien @liamocn this is something that eventually would be displayed on a map on the portal I guess but since I have to produce that summary once in a while for Jessica, I figured it would be useful to have this script (and maybe other aggregation scripts) in the repo.

Let me know if you disagree having that here. I can always create a repo for myself with those aggregated scripts. Thanks!

emmetaobrien commented:
Seems like a good idea to me, generally.  One minor thing: the header line is out of alignment with the data in your example output.

cmadjar commented:
@emmetaobrien good catch!! I added the Principal Investigator value to the rows but forgot to add it to the header row. Fixed in the script now. Thank you!!

cmadjar commented:
oooops. Just realized that the branches I created were created on the conp organization, not cmadjar... Will resubmit them properly and delete the conp branches I created by mistake... C'est vendredi ;)

cmadjar commented:
New PR: https://github.com/CONP-PCNO/conp-dataset/pull/426
